great_expectations/expectations/metrics/table_metrics/table_column_types.py

```diff
@@ -83,8 +88,13 @@ def _spark(
 
 
 def _get_sqlalchemy_column_metadata(engine, batch_data: SqlAlchemyBatchData):
-    table_selectable = batch_data.source_table_name or batch_data.selectable.name
-    schema_name = batch_data.source_schema_name or batch_data.selectable.schema
+    # if a custom query was passed
+    if isinstance(batch_data.selectable, TextClause):
+        table_selectable = batch_data.selectable
+        schema_name = None
+    else:
+        table_selectable = batch_data.source_table_name or batch_data.selectable.name
+        schema_name = batch_data.source_schema_name or batch_data.selectable.schema
```

Comment:
Could we add type hints here to make this clearer? Is the `table_selectable` on L96 generally a consistent type?

Reply:
Yes, that makes sense. Typically `table_selectable` would be a string representing the name of the table. I will make this change.